civitai-downloader-v2/src/core/search/strategy.py

```
@@ -0,0 +1,546 @@
+                prev_page=metadata_raw.get('prevCursor')   # For cursor-based pagination
+            )
+            
+            # Update statistics
+            response_time = time.time() - start_time
+            self._update_stats(len(results), response_time, success=True)
+            
+            return results, metadata
+            
+        except Exception as e:
+            self._update_stats(0, time.time() - start_time, success=False)
+            raise Exception(f"Search failed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)